packages/cli/src/models/local/UnpackController.ts

```diff
@@ -0,0 +1,72 @@
+import path from 'path';
+import request from 'request-promise-native';
+import fs from 'fs-extra';
+
+import { Logger } from 'zos-lib';
+
+import stdout from '../../utils/stdout';
+import patch from '../../utils/patch';
+import cp from '../../utils/cp';
+
+const ora = patch('ora');
+const simpleGit = patch('simple-git/promise');
+
+interface KitConfig {
+  ignore: string[],
+  hooks: Object,
+}
+
+export default class UnpackController {
+  public async unpack(url: string, workingDirPath: string = '', config: KitConfig): Promise<void | never> {
+    if (!url) throw Error('A url must be provided.');
+    if (!config) throw Error('A config must be provided.');
+
+    const { exec } = cp;
+    const { readdir, remove, pathExists } = fs;
+
+    // because zos always spawns '.zos.lock' file
+    if ((await readdir(workingDirPath)).length > 1) throw Error('The directory must be empty');
+
+    try {
+      let spinner = ora(`Downloading kit from ${url}`).start();
+      const git = simpleGit(workingDirPath);
+      await git.init();
+      await git.addRemote('origin', url);
+      await git.pull('origin', 'stable');
+      spinner.succeed();
+
+      spinner = ora(`Unpacking kit`).start();
+      // always delete .git folder 
+      config.ignore.push(`.git`);
+      // delete all files/folders from ignore
+      const paths = config.ignore.map(file => path.join(workingDirPath, file));
+      const exists = await Promise.all(paths.map(path => pathExists(path)));
+      await Promise.all(exists.map((ext, i)=> ext ? remove(paths[i]) : null ).filter(Boolean));
+      
+      // run kit commands like `npm install`
+      await exec(config.hooks['post-unpack']);
+      spinner.succeed();
+
+      stdout(`Done. Amazing!`);
+      stdout(`Please, continue at ${url}`);
```

Comment:
Shouldn't the kit have the responsibility of showing the next steps? Maybe we could include a "description" or "next steps" field in the `kit.json` that is shown here, or just make it part of the `post-unpack`.

Reply:
My first implementation was like that and then I realized it is wrong.
I was thinking about it for quite a bit. Problem with showing next step or description in terminal are:
* Terminal has limited formatting support 
* It is hard to go back to terminal after running more commands and `clear` command would wipe it out
I believe that the way other projects do it is wrong. The print long list of commands which is gone after 10 minutes of working in terminal and there is no way to go back.
That is why I decided to focus to bringing user to the kit readme file which can be opened with rich content and sustained for longer time.
Link in terminal is clickable in Visual Studio Code or can be copied.

Reply:
You make a good point. Nevertheless, I'm not sure they are mutually exclusive: a project may want to include some basic next steps, as well as redirecting to the README. 

My main concern with the redirect is that only a fraction of the users will actually click on the link. As an example, OZ recently got an issue opened from a very good dev, that was solved just by clicking on a link included on the error msg. So, it would be good to include the very basic points right in the terminal.

Reply:
I agree we need both. It totally makes sense for to provide kit's developers a way to show message after unpacking. I would say we should both with heavy and rich guide at github and short and to the point instructions at terminal.